scandium-core/src/main/java/org/eclipse/californium/scandium/auth/ApplicationLevelInfoSupplier.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ *******************************************************************************/
+
+
+package org.eclipse.californium.scandium.auth;
+
+import java.security.Principal;
+import java.util.Set;
+
+/**
+ * A strategy for retrieving additional (application level) information about an authenticated client.
+ */
+public interface ApplicationLevelInfoSupplier {
+
+    /**
+     * Gets additional information about an authenticated client.
+     * 
+     * @param clientIdentity The client identity.
+     * @return The additional information for the client.
+     */
+    Set<Principal> getInfo(Principal clientIdentity);
```

Comment:
The `Set`-solution may also fit for issue #974 .

Is there any intention to access a specific `Principal` in `Set`?
Any thoughts about using a `Map<String, Principal>` ?